gen2_demo.py

```diff
@@ -171,12 +220,29 @@ def fps(self):
         return self.frame_cnt / (self.timestamp - self.start)
 
 
-def create_pipeline(use_camera, use_hq, nn_pipeline=None):
+def create_pipeline(use_camera, use_hq, use_depth, nn_pipeline=None):
```

Comment:
Seeing how much this function has grown, can we make it like a `PipelineManager`?
that would be a class with functions like `createDepth` or `createLeftCam` so that we can make it more flexible?
This relates also to the comment above about separating the logic in this script